### PR TITLE
[DOCS] remove unsupported actions

### PR DESCRIPTION
**Pull request by klavavej**
resolves https://greatexpectations.atlassian.net/browse/DOC-844 - removes unsupported actions from the API docs (opsgenie, pagerduty, SNS)

I'm leaving `UpdateDataDocsAction` and `ValidationAction` in place as I assume those are supported and the ticket was about third party integrations that aren’t currently supported. Let me know if I’m wrong though

- [X] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [X] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `ruff format` + `ruff check`)
- [X] Appropriate tests and docs have been updated

For more information about contributing, visit our [community resources](https://docs.greatexpectations.io/docs/core/introduction/community_resources#contribute-code-or-documentation).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

### CODE REVIEW

File: great_expectations/checkpoint/actions.py [hunk 1]
```diff
@@ -352,7 +352,6 @@ def _send_slack_notification(self, payload: dict) -> dict:
         return {"slack_notification_result": slack_notif_result}
 
 
-@public_api
```
Comment (billdirks) [inline great_expectations/checkpoint/actions.py:355]: Are these actions no longer GX support or no longer community supported either?

That is, do we want community supported actions in our public docs and labeled as such or not appearing at all.

If these aren't community supported, can we delete the actions altogether?

Comment (klavavej) [reply great_expectations/checkpoint/actions.py:355]: > Are these actions no longer GX support or no longer community supported either?

These actions are unsupported by GX _and_ unsupported by the community (source of truth for integration support is https://docs.greatexpectations.io/docs/application_integration_support#integrations) 

> If these aren't community supported, can we delete the actions altogether?

Yes, but that's outside the scope of this docs PR. I filed https://greatexpectations.atlassian.net/browse/CORE-600 for this and product/eng can then prioritize as y'all see fit (I'm not sure how impactful the tech debt is / if it's worth your time to clean it up or easier to just leave it in place in case product decides we should support these integrations in the future)

Comment (billdirks) [reply great_expectations/checkpoint/actions.py:355]: We should always remove dead code.

File: docs/docusaurus/static/_redirects [hunk 2]
```diff
@@ -73,7 +73,7 @@ http://legacy.016.docs.greatexpectations.io/* http://docs.greatexpectations.io 3
 /en/latest/guides/how_to_guides/miscellaneous/how_to_template.html https://legacy.docs.greatexpectations.io/en/latest/guides/how_to_guides/miscellaneous/how_to_template.html
 /en/latest/guides/how_to_guides/miscellaneous/how_to_add_and_test_a_new_sqlalchemydataset_class.html https://legacy.docs.greatexpectations.io/en/latest/guides/how_to_guides/miscellaneous/how_to_add_and_test_a_new_sqlalchemydataset_class.html
 /en/latest/guides/how_to_guides/miscellaneous/how_to_use_official_docker_images.html https://legacy.docs.greatexpectations.io/en/latest/guides/how_to_guides/miscellaneous/how_to_use_official_docker_images.html
-/en/latest/guides/how_to_guides/miscellaneous/how_to_setup_opsgenie_alert_notifications.html https://legacy.docs.greatexpectations.io/en/latest/guides/how_to_guides/miscellaneous/how_to_setup_opsgenie_alert_notifications.html
+/en/latest/guides/how_to_guides/miscellaneous/how_to_setup_opsgenie_alert_notifications.html /docs/0.18/reference/api/checkpoint/opsgeniealertaction_class/
```
Comment (billdirks) [inline docs/docusaurus/static/_redirects]: Why aren't we completely removing opgenie here if it is not supported? Is it because the code still exists? When we remove that code, will we have to remove this code and update this file again?

File: tests/checkpoint/test_checkpoint.py [hunk 3]
```diff
@@ -651,12 +649,8 @@ def test_checkpoint_sorts_actions(self, validation_definition: ValidationDefinit
         Pydantics and mocks.
         Ideally, this would be tested through the public `run()` method.
         """
-        pd_action = PagerdutyAlertAction(
-            name="my_pagerduty_action", api_key="api_key", routing_key="routing_key"
-        )
-        og_action = OpsgenieAlertAction(name="my_opsgenie_action", api_key="api_key")
         data_docs_action = UpdateDataDocsAction(name="my_docs_action")
-        actions: List[CheckpointAction] = [pd_action, og_action, data_docs_action]
+        actions: List[CheckpointAction] = [data_docs_action]
```
Comment (billdirks) [inline tests/checkpoint/test_checkpoint.py]: This test is suppose to verify sorting actions. If there is only 1 action, this test is no longer verifying this. Could you add actions to replace the ones you removed?

File: docs/sphinx_api_docs_source/public_api_excludes.py [hunk 4]
```diff
@@ -827,4 +827,19 @@
         name="get_batch_parameters_keys",
         filepath=pathlib.Path("great_expectations/datasource/fluent/sql_datasource.py"),
     ),
+    IncludeExcludeDefinition(
```
Comment (billdirks) [inline docs/sphinx_api_docs_source/public_api_excludes.py:830]: If we were to remove the unsupported actions would we have to undo this change and remove these?

**Review by billdirks — CHANGES_REQUESTED** [review]
Thanks for working towards removing unsupported docs. It looks like the updated test no longer tests what the test name claims it does.

File: docs/docusaurus/static/_redirects (hunk 2 above)
Comment (billdirks) [reply docs/docusaurus/static/_redirects]: That is, if we need to remove it later, could you add this to the ticket you filed, CORE-600?

File: docs/sphinx_api_docs_source/public_api_excludes.py (hunk 4 above)
Comment (billdirks) [reply docs/sphinx_api_docs_source/public_api_excludes.py:830]: If so, could you add it as a task to do in CORE-600?

File: docs/docusaurus/static/_redirects (hunk 2 above)
Comment (klavavej) [reply docs/docusaurus/static/_redirects]: hmmm good point! I forgot this part of our support posture -https://docs.greatexpectations.io/docs/application_integration_support#gx-versions - I'll revise this redirect

File: tests/checkpoint/test_checkpoint.py (hunk 3 above)
Comment (klavavej) [reply tests/checkpoint/test_checkpoint.py]: I'll try! But I'm definitely out of my comfort zone here so if the update I make doesn't do the trick I'd love to pair on this

File: docs/sphinx_api_docs_source/public_api_excludes.py (hunk 4 above)
Comment (klavavej) [reply docs/sphinx_api_docs_source/public_api_excludes.py:830]: > If we were to remove the unsupported actions would we have to undo this change and remove these?

I assume yes, but I'm not 100% confident in my understanding of how the api docs get built

> If so, could you add it as a task to do in CORE-600?

yes, will do!

File: great_expectations/checkpoint/actions.py (hunk 1 above)
Comment (billdirks) [reply great_expectations/checkpoint/actions.py:355]: Thinking about this more, maybe we need to leave the code in for semantic versioning. Removing the public api decorator will necessitate a minor version bump.

File: tests/checkpoint/test_checkpoint.py (hunk 3 above)
Comment (billdirks) [reply tests/checkpoint/test_checkpoint.py]: Thanks for doing this! Happy to pair.

**Review by billdirks — APPROVED** [review]
LGTM, thanks!

File: docs/sphinx_api_docs_source/public_api_excludes.py (hunk 4 above)
Comment (tyler-hoffman) [reply docs/sphinx_api_docs_source/public_api_excludes.py:830]: @klavavej the api docs build is complex. I'm happy to chat about it.
